lib/std/Build/CompileStep.zig

```diff
@@ -1555,7 +1555,9 @@ fn make(step: *Step) !void {
                     try zig_args.append("--test-cmd");
                     try zig_args.append(bin_name);
                     try zig_args.append("--test-cmd");
-                    try zig_args.append("--dir=.");
+                    try zig_args.append("--mapdir=/::.");
+                    try zig_args.append("--test-cmd");
+                    try zig_args.append("--mapdir=/tmp::/tmp");
```

Comment:
We can prevent adding those preopens by manually creating an instance of `Preopen` like so:

```zig
const wasi_preopens: Preopens = .{
    .names = &.{
        "stdin",
        "stdout",
        "stderr",
        ".",
        "/tmp",
        ...etc
    },
};
```
By simply hardcoding some paths we can then call `find` and `findDir` and test for the expected results. This prevents us from having to pass the temporary directory as a preopen for all users while still allowing to test the use-case: finding the correct preopen based on a prefix or exact match.

Reply:
🤯

Reply:
ok I did that, but I now the test is panicking on `fs.Dir.makeOpenPath`. I guess the reason is that we're now in an impossible situation where the preopen exists in the data structure, but wasmtime has not actually preopened that directory. 

In the test first I create a directory under `/tmp` by using the exact `find()` routine, then I write to it, and then I expect to find the result using `findDir`

```
37/40 test.Preopens.findDir... reached unreachable codeError: failed to run main module `/Users/evacchi/Devel/github.com/ziglang/zig/zig-cache/o/15be5938debf463f54cb5df5a921f379/test.wasm`

Caused by:
    0: failed to invoke command default
    1: error while executing at wasm backtrace:
           0: 0x437b - <unknown>!os.abort
           1: 0x1d2c - <unknown>!builtin.default_panic
           2: 0x65a53 - <unknown>!os.mkdiratWasi
           3: 0x658c3 - <unknown>!os.mkdirat
           4: 0x65807 - <unknown>!fs.Dir.makeDir
           5: 0x65d0c - <unknown>!fs.Dir.makePath
           6: 0x661ba - <unknown>!fs.Dir.makeOpenPath
           7: 0x67089 - <unknown>!fs.wasi.test.Preopens.findDir
           8:  0xd4e - <unknown>!test_runner.main
           9:  0x698 - <unknown>!_start
       note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information
    2: wasm trap: wasm `unreachable` instruction executed
error: the following test command failed with exit code 134:
wasmtime --dir=. /Users/evacchi/Devel/github.com/ziglang/zig/zig-cache/o/15be5938debf463f54cb5df5a921f379/test.wasm
```

Reply:
Sorry for being unclear. My suggestion is to hardcode the `Preopens` list while only relying on `find` and `findDir` functionality, removing all instances where additional paths/directories are created. This would still test the functionality of your implementation.

i.e.
If you were to have a preopen `/tmp` I expect it would return null for `find("/tmp/file.txt")` but would return a result for `findDir("/tmp/file.txt")` and for `find("/tmp")`.

Reply:
that's right. Sure I can update it that way, but the reason why the test was coded that way was to make sure a longer match would yield the other preopen :)

i.e. : `/tmp/foo` yields preopen `/tmp` sub-path `foo`, not preopen `/` sub-path `tmp/foo` (which is currently what happens today with any API that defaults to cwd)